drivers/SmartThings/zigbee-switch/src/frient/init.lua

```diff
@@ -0,0 +1,209 @@
+-- Copyright 2025 SmartThings
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+local capabilities = require "st.capabilities"
+local clusters = require "st.zigbee.zcl.clusters"
+local constants = require "st.zigbee.constants"
+local log = require "log"
+local util = require "st.utils"
+local configurationMap = require("configurations")
+
+local SimpleMetering = clusters.SimpleMetering
+local ElectricalMeasurement = clusters.ElectricalMeasurement
+local Alarms = clusters.Alarms
+
+local VOLTAGE_MEASUREMENT_MULTIPLIER_KEY = "_voltage_measurement_multiplier"
+local VOLTAGE_MEASUREMENT_DIVISOR_KEY    = "_voltage_measurement_divisor"
+local CURRENT_MEASUREMENT_MULTIPLIER_KEY = "_current_measurement_multiplier"
+local CURRENT_MEASUREMENT_DIVISOR_KEY    = "_current_measurement_divisor"
+
+local FRIENT_SMART_PLUG_FINGERPRINTS = {
+    { mfr = "frient A/S", model = "SPLZB-131" },
+    { mfr = "frient A/S", model = "SPLZB-132" },
+    { mfr = "frient A/S", model = "SPLZB-134" },
+    { mfr = "frient A/S", model = "SPLZB-137" },
+    { mfr = "frient A/S", model = "SPLZB-141" },
+    { mfr = "frient A/S", model = "SPLZB-142" },
+    { mfr = "frient A/S", model = "SPLZB-144" },
+    { mfr = "frient A/S", model = "SPLZB-147" },
+    { mfr = "frient A/S", model = "SMRZB-143" },
+    { mfr = "frient A/S", model = "SMRZB-153" },
+    { mfr = "frient A/S", model = "SMRZB-332" },
+    { mfr = "frient A/S", model = "SMRZB-342" },
+}
+
+local POWER_FAILURE_ALARM_CODE = 0x03
+
+
+local function alarm_report_handler(driver, device, zb_rx)
+    local alarm_status = zb_rx.body.zcl_body
+    log.error("Received alarm report:"..util.stringify_table(zb_rx.body.zcl_body, nil, true))
+    if ((alarm_status.cluster_identifier.value == SimpleMetering.ID) and (alarm_status.alarm_code.value == POWER_FAILURE_ALARM_CODE)) then
+        device.thread:call_with_delay(2, function(d)
+            device:emit_event_for_endpoint(zb_rx.address_header.src_endpoint.value, capabilities.powerSource.powerSource.unknown())
+        end
+        )
+    end
+end
+
+local function device_added(driver, device)
+end
+
+-- Handler for voltage measurement
+local function voltage_measurement_handler(driver, device, value, zb_rx)
+    local raw_value  = value.value
+    -- By default emit raw value
+    local multiplier = device:get_field(VOLTAGE_MEASUREMENT_MULTIPLIER_KEY) or 1
+    local divisor    = device:get_field(VOLTAGE_MEASUREMENT_DIVISOR_KEY) or 1
+
+    if divisor == 0 then
```

Comment:
If this is a concern, I would put the check where you write the value to `VOLTAGE_MEASUREMENT_DIVISOR_KEY`.